File: sprint1/By languages/JS/lesson7code5.js
function eratosthenes_effective(n) {
    const numbers = new Array(n + 1).fill(true);
    numbers[0] = numbers[1] = false;
    for (let num = 2; num < n; num++) {
        if (numbers[num]) {
            for (let j = num * num; j <= n; j += num) {
                numbers[j] = false;
            }
        }
    }
    return numbers;
}

File: sprint1/By languages/swift/lesson7code5.swift
func eratosthenesEffective(_ n: Int) -> [Bool] {
    var numbers = Array(repeating: true, count: n + 1)
    numbers[0] = false
    numbers[1] = false
    
    for num in 2..<n {
        if numbers[num] {
            var j = num * num
            while j <= n {
                numbers[j] = false
                j += num
            }
        }
    }
    
    return numbers
}